sway-core/src/semantic_analysis/type_check_context.rs

```diff
@@ -1066,6 +1136,7 @@ impl<'a> TypeCheckContext<'a> {
         impl_span: &Span,
         trait_decl_span: Option<Span>,
         is_impl_self: bool,
+        is_extending: bool,
```

Comment:
Can we get enums or something that makes callsites more explicit for these two? `is_extending` is also not very clear, consider renaming it or adding some documentation as to what it means.

Reply:
Changed to use enums `IsImplSelf` and `IsExtendingExistingImpl`.